src/main/ruby/core/truffle/cext.rb

```diff
@@ -77,12 +77,21 @@ def self.library_in_search_path_or_default(lib, search_paths)
       lib
     end
 
-    def self.standardize_lib_name(lib)
-      if lib.start_with?('lib') or lib.start_with?('/')
-        lib
-      else
-        "lib#{lib}.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
-      end
+    def self.standardize_lib_name(lib_name)
+      require 'ffi'
+      extend FFI::Library
+
+      # Take a first pass at normalizing the library name much like the linker would normally do for us.
+      # Since we ultimately call `dlopen` to open the library, the name must be in a form that `dlopen` can
+      # handle. This pass will, for example, map "jpeg" -> "libjpeg.so".
+      normalized_lib_name = FFI.map_library_name(lib_name)
+
+      # Mapping a library name will only handle hard-coded cases accurately. Otherwise, it takes a best effort approach
```

Comment:
`hard-coded cases` is a bit vague, it's really only `libc`.

Reply:
There's no reason we can't expand that. We probably should have it for `libm`. But it's an implementation detail of the method being called. The claim that only hard-coded cases are handled is wrong though. The generic pattern works in many cases.